lib/explorer/series.ex

```diff
@@ -16,6 +16,8 @@ defmodule Explorer.Series do
     * `:time` - Time type that unwraps to `Elixir.Time`
     * `{:list, dtype}` - A recursive dtype that can store lists. Examples: `{:list, :integer}` or
       a nested list dtype like `{:list, {:list, :integer}}`.
+    * `{:struct, %{key => dtype}}` - A recursive dtype that can store structs. Examples:
+      `{:struct, %{"a" => :integer}}` or a nested struct dtype like `{:struct, %{"a" => {:struct, %{"b" => :integer}}}}`.
```

Comment:
I think we could mention that `:struct` is not related to Elixir structs, but to [Arrow](https://arrow.apache.org/docs/format/Columnar.html#struct-layout)/Polars structs.

Reply:
That's nice! I'll make an edit adding this later.